Adaptors/S3/src/UploadMultiPartHelper.cs

```diff
@@ -0,0 +1,112 @@
+// This file is part of the ArmoniK project
+// 
+// Copyright (C) ANEO, 2021-2023. All rights reserved.
+// 
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published
+// by the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+// 
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY, without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+// 
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+using Amazon.S3.Model;
+
+namespace ArmoniK.Core.Adapters.S3;
+
+internal class UploadMultiPartHelper
+{
+  private const long MinPartSize  = 5 * 1024 * 1024; // 5 MiB
+  private const int  MaxPartCount = 10000;
+  private const long MaxPartSize  = 5L * 1024 * 1024 * 1024; // 5 GiB
+
+  public static async Task<List<UploadPartRequest>> PreparePartRequestsAsync(string                                 bucketName,
```

Comment:
Instead of returning a Task of list of requests, I think it should be better to return a IAsyncEnumerable of requests.